### ISSUE TITLE
transferable objects

### ISSUE THREAD

**Issue by francescoagati** [josdejong/workerpool#3]
Hi,
there is a support for transferable objects?

**Comment by josdejong** [comment]
That's not supported right now but would be a valuable addition. Feel free to contribute if you like :)

**Comment by blake-regalia** [comment]
I think this could be supported in node.js environment using [shm-typed-array](https://github.com/ukrbublik/shm-typed-array)...

**Comment by ahkohd** [comment]
Hmm, I have this idea in my head. 🤷

Say we have a dedicated web worker **(X)** for holding transferable objects for the moment.

1. You invoke `pool.exec()`
2. The transferable object you passed instantly gets transferred to worker **X** which also keeps track of which job owns the transferable object.
3. When your job is ready for processing by another web worker say (Y).
4. `Worker Y` sends a message to `worker X` to ask if you have any transferable object intended for the job its about to run.
5. If yes, worker X transfers the transferable object to Y.
6. Problem solved.

**Comment by josdejong** [comment]
I guess that's indeed a good description of what we want `workerpool` to do: hold the object/array, pass to the worker which is going to process it, and in the end transfer the results back to the main thread. I'm not sure if there is need to keep track of a job ID or not.

**Comment by ahocevar** [comment]
In anology to [what `threadpool-js` does](https://github.com/andywer/threadpool-js#support-for-transferable-objects), the API could be to pass the array of transferable objects as additional argument to the `exec()` method:
```js
pool.exec("mythread", {hash: "sha512", data: myUint8Array}, [myUint8Array.buffer]);
```

**Comment by josdejong** [comment]
That looks like a neat API indeed. I guess there should also be a way to return a transferable object. Any suggestions for that?

**Comment by josdejong** [comment]
That makes sense. Your proposal looks good to me 👍

**Comment by josdejong** [comment]
Transferring objects is now available in `v6.4.0`
